src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -441,29 +441,18 @@ async def list_current_user_dataset_records(
 
     await authorize(current_user, DatasetPolicyV1.get(dataset))
 
-    if metadata.metadata_parsed or sort_by_query_param:
-        records, total = await _filter_records_using_search_engine(
-            db,
-            search_engine,
-            dataset=dataset,
-            parsed_metadata=metadata.metadata_parsed,
-            limit=limit,
-            offset=offset,
-            user=current_user,
-            response_statuses=response_statuses,
-            include=include,
-            sort_by_query_param=sort_by_query_param or {RecordSortField.inserted_at.value: "asc"},
-        )
-    else:
-        records, total = await datasets.list_records_by_dataset_id(
-            db,
-            dataset_id,
-            current_user.id,
-            include=include,
-            response_statuses=response_statuses,
-            offset=offset,
-            limit=limit,
-        )
+    records, total = await _filter_records_using_search_engine(
+        db,
+        search_engine,
+        dataset=dataset,
+        parsed_metadata=metadata.metadata_parsed,
+        limit=limit,
+        offset=offset,
+        user=current_user,
+        response_statuses=response_statuses,
+        include=include,
+        sort_by_query_param=sort_by_query_param or {RecordSortField.inserted_at.value: "asc"},
```

Comment:
We should move `{RecordSortField.inserted_at.value: "asc"}` to a constant. 

Something like:

```suggestion
        sort_by_query_param=sort_by_query_param or LIST_DATASET_RECORDS_SORT_BY,
```